src/main/java/com/ge/predix/uaa/token/lib/ZacTokenService.java

```diff
@@ -16,64 +16,113 @@
-                    .getForEntity(trustedIssuersURL, TrustedIssuers.class);
-            tokenServices = super.createFastTokenService(responseEntity.getBody().getTrustedIssuerIds());
+            OAuth2ClientHttpRequestInterceptor interceptor = getOAuth2ClientHttpRequestInterceptor();
+            ResponseEntity<TrustedIssuers> response = RestClient.builder()
+                                                                .requestInterceptor(interceptor)
+                                                                .defaultStatusHandler(new DefaultResponseErrorHandler())
+                                                                .build().get().uri(trustedIssuersURL).retrieve()
+                                                                .toEntity(TrustedIssuers.class);
+            tokenServices = super.createFastTokenService(
+                Objects.requireNonNull(response.getBody()).getTrustedIssuerIds());
         } catch (Exception e) {
-            LOGGER.error("Failed to get trusted issuers from: " + trustedIssuersURL);
-            LOGGER.error(e.getMessage());
+            LOGGER.error("Failed to get trusted issuers from: {}", trustedIssuersURL, e);
             throw e;
         }
+        LOGGER.debug("Created FastTokenServices for zone: {}", zoneId);
         return tokenServices;
     }
 
-    public void setOauth2RestTemplate(final RestOperations oauth2RestTemplate) {
-        this.oauth2RestTemplate = oauth2RestTemplate;
+    /**
+     * Creates an OAuth2ClientHttpRequestInterceptor for the client registration.
+     *
+     * @return the OAuth2ClientHttpRequestInterceptor instance
+     */
+    private OAuth2ClientHttpRequestInterceptor getOAuth2ClientHttpRequestInterceptor() {
+        var clientRegistrationRepository = new InMemoryClientRegistrationRepository(clientRegistration);
+        var clientService = new InMemoryOAuth2AuthorizedClientService(clientRegistrationRepository);
+        var authorizedClientManager =
+            new AuthorizedClientServiceOAuth2AuthorizedClientManager(clientRegistrationRepository, clientService);
+        var interceptor = new OAuth2ClientHttpRequestInterceptor(authorizedClientManager);
+        interceptor.setClientRegistrationIdResolver(request -> clientRegistration.getRegistrationId());
+        return interceptor;
     }
 
-    @Required
-    public void setZacUrl(final String zacUrl) {
-        this.zacUrl = zacUrl;
+    /**
+     * Sample Client registration snippet
+     * ClientRegistration.withRegistrationId("zacUaaClientDetails")
+     * .authorizationGrantType(AuthorizationGrantType.CLIENT_CREDENTIALS)
+     * .clientId(uaaAdminClientId)
+     * .clientSecret(uaaAdminClientSecret)
+     * .clientAuthenticationMethod(ClientAuthenticationMethod.CLIENT_SECRET_POST)
+     * .tokenUri(uaaTokenUri)
+     * .build()
+     */
+    public void setZacUaaClientDetails(ClientRegistration clientRegistration) {
```

Comment:
if `clientRegistration` is mandatory field, can we move it to constructor arg without major refactoring? and where we are calling `setZacUaaClientDetails(ClientRegistration clientRegistration)` method?

Reply:
Moved now